src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -903,6 +903,28 @@ impl ScalarFunc {
             Ok(Some(Cow::Borrowed(b"")))
         }
     }
+
+    #[inline]
+    pub fn instr_binary<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let s = try_opt!(self.children[0].eval_string_and_decode(ctx, row));
```

Comment:
We shouldn't decode the string with UTF-8 since it is a binary and may not be encoded by UTF-8.